app/controllers/api/v1/offerings_controller.rb

```diff
@@ -24,6 +24,35 @@ def show
     render :json => offering_api.to_json, :callback => params[:callback]
   end
 
+  # PUT /portal_offerings/1
+  def update
+    offering = Portal::Offering.find(params[:id])
+    authorize offering
+    offering.update_attributes!(params.permit(:active, :locked))
+    if params[:position]
+      new_pos = params[:position].to_i
+      class_offerings = offering.clazz.teacher_visible_offerings
```

Comment:
So, I've left `teacher_visible_offerings` as this is pretty much the scope available to the teacher (and everyone else). I guess we could rename it to `non_archived_offerings` if it makes sense to you. Let me know if that's okay or if you still want to handle archived offerings in reordering (it will complicate PortalPages code a lot, as we don't even list those offerings in API answers).

Note that actually it's not so hard to test API and I've already added tests of reordering before:
https://github.com/concord-consortium/rigse/blob/27a25f591866a3bb5013e3a4ee1cf3d3d112b2bd/spec/controllers/api/v1/offerings_controller_spec.rb#L524-L546